### PR TITLE
#6947: restricted URL/extension console popover

### LINKED ISSUE #6947: Show popover on restricted URLs
## User Story

As a PixieBrix end-user, I want to see a popover when I try to access PixieBrix on a restricted URL so I can figure out how to access the PixieBrix sidebar

## Context

https://www.notion.so/pixiebrix/Improve-Browser-Toolbar-UX-on-Extension-Console-and-non-scriptable-URLs-1f3a0f2452d74653823201103d1c4d22?pvs=4

### PR DESCRIPTION
**Pull request by twschiller**
## What does this PR do?

- Closes #6947
- Shows an information popover when clicking the browser action button on restricted URLs
- Shows an information popover when clicking the browser action button on the Extension Console

## Remaining Work

- [x] Fix type issue: https://github.com/pixiebrix/pixiebrix-extension/actions/runs/6949528190/job/18907838003?pr=6948. It's not clear to me why that would fail given its in an unrelated file
- [x] Investigate potential slowdown in sidebar toggling - seems to be transient/unrelated

## Discussion

- I added the Page Editor welcome link. We might chose to exclude that if the user is a non-developer/restricted user

## Demo

- https://www.loom.com/share/7144f464b062435498e730b15b467057

## Checklist

- [ ] Add tests: we'll need to add a rainforest test here
- [x] New files added to `src/tsconfig.strictNullChecks.json` (if possible)
- [x] Designate a primary reviewer: @grahamlangford

### CODE REVIEW

File: src/telemetry/events.ts [hunk 1]
```diff
@@ -24,6 +24,8 @@ export const Events = {
   BRICK_ADD: "BrickAdd",
   BRICK_DELETE: "BrickDelete",
 
+  BROWSER_ACTION_RESTRICTED_URL: "BrowserActionRestrictedUrl",
```
Comment (grahamlangford) [inline src/telemetry/events.ts:27]: 💜

File: src/tinyPages/restrictedUrlPopup.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import "@/extensionContext";
```
Comment (grahamlangford) [inline src/tinyPages/restrictedUrlPopup.tsx:18]: This file was not yet strict null checked, which is what's causing the failure. I just pushed a small commit to handle it.

**Review by grahamlangford — APPROVED** [review]

**Comment by twschiller** [comment]
@fregante could you confirm the active tab URL tracking looks OK? Seems to be working well

**Comment by fregante** [comment]
Mostly fine, I caught a couple more situations in https://github.com/pixiebrix/pixiebrix-extension/pull/6954

**Pull request merged**